tinymce-language-selector/plugin.js

```diff
@@ -6,17 +6,20 @@ tinymce.PluginManager.add('language', function(editor) {
   const replaceText = (lang) => {
     const selectedNode = editor.selection.getNode();
     const newText = editor.selection.getContent({ format: 'text' });
-    const contents = newText || '&#65279';
+    const contents = newText || '&#65279'; // use zero-width character as placeholder if no existing text
     const newSpanText = lang === BROWSER_DEFAULT ? `<span id="new_span">${contents}</span>` : `<span lang="${lang}" id="new_span">${contents}</span>`;
     // may be in the middle of the span, so split it into two
     if (selectedNode.nodeName === 'SPAN') {
       if (selectedNode.lang) {
-        editor.execCommand('mceReplaceContent', false, `</span>${newSpanText}<span lang="${selectedNode.lang}">`);
+        editor.selection.setContent(''); // delete any existing text
```

Comment:
This should just be moved outside of the if conditions if we are going to call it in every case.